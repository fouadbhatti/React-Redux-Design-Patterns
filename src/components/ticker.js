import React, { memo } from 'react';

export function Ticker({dailyChangePerc, lastPrice, volume, high, low}) {
  return (
    <div>
      <div>Daily Change: {dailyChangePerc}</div>
      <div>Last Price: {lastPrice}</div>
      <div>Volume: {volume}</div>
      <div>High: {high}</div>
      <div>Low: {low}</div>
    </div>
  );
}

export default memo(Ticker);