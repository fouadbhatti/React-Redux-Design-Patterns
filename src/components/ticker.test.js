import React from 'react';
import renderer from 'react-test-renderer';
import { Ticker } from './ticker';

describe('Ticker Component Test', () => {
  it('renders as expected', () => {
    const componentTree = renderer.create(
      <Ticker
        dailyChangePerc={0.1}
        lastPrice={100}
        volume={1000}
        high={5500}
        low={3000}
      />)
    .toJSON();

    expect(componentTree).toMatchSnapshot();
  });
});