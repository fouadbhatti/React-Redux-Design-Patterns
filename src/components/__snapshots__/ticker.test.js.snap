// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ticker Component Test renders as expected 1`] = `
<div>
  <div>
    Daily Change: 
    0.1
  </div>
  <div>
    Last Price: 
    100
  </div>
  <div>
    Volume: 
    1000
  </div>
  <div>
    High: 
    5500
  </div>
  <div>
    Low: 
    3000
  </div>
</div>
`;
